packages/utils/src/route.js

```diff
@@ -41,29 +41,26 @@ export const flatRoutes = function flatRoutes (router, fileName = '', routes = [
   return routes
 }
 
-function cleanChildrenRoutes (routes, isChild = false, routeNameSplitter = '-', trailingSlash) {
-  let start = -1
+function cleanChildrenRoutes (routes, isChild = false, routeNameSplitter = '-', trailingSlash, namePrefix) {
   const regExpIndex = new RegExp(`${routeNameSplitter}index$`)
   const routesIndex = []
   routes.forEach((route) => {
     if (regExpIndex.test(route.name) || route.name === 'index') {
       // Save indexOf 'index' key in name
-      const res = route.name.split(routeNameSplitter)
-      const s = res.indexOf('index')
-      start = start === -1 || s < start ? s : start
+      const res = route.name.replace(namePrefix, '').split(routeNameSplitter)
```

Comment:
Do you want to only remove prefix if it's at the beginning of the name? In that case, it would probably make sense to make this replace more robust so that it doesn't try to replace anywhere within the name.